### ISSUE TITLE
Parallel serialization tutorial.

### ISSUE THREAD

**Issue by bangerth** [dealii/dealii#17270]
step-83 (see #12429) only shows how to serialize a sequential program. That is undesirable, but expedient for getting the program merged in time for the current release.

Here are some follow-up steps we can discuss for the next release:
* We make step-83 parallel, as originally planned, by showing how to use `SolutionTransfer` for attaching solution vectors and particles to a triangulation. In fact, @peterrum 's patch in https://github.com/dealii/dealii/pull/12429#issuecomment-2226063223 almost does all of that already.
* We take another program that's already parallel and show how to serialize that. @blaisb in https://github.com/dealii/dealii/pull/12429#issuecomment-2232211665 suggests to take step-68.
* We rethink the whole approach with `SolutionTransfer` and see whether we can squeeze that into `operator&` somehow. I have not thought this through, but it's conceivable that at least for `ParticleHandler`, one could do something like
```
  template <int dim, int spacedim>
  template <class Archive>
  inline void
  ParticleHandler<dim, spacedim>::serialize(Archive &ar, const unsigned int)
  {
    ar &global_number_of_particles &global_max_particles_per_cell
        &next_free_particle_index;

   if constexpr (Archive is an output stream)
   {
      create a Particles::SolutionTransfer object, connect the current object to it, and register it with the triangulation
   } else {
      create a Particles::SolutionTransfer object, connect the current object to it, and register it with the triangulation
   }
}
```
In other words, I wonder whether it would be possible to just attach oneself to the triangulation and hide all the details of setting up helper objects.

(I recognize that that last option may not work very well. It means that every process would have to call serialization, which we probably don't want to do -- not the least because then not necessarily the same number of processes serialize as deserialize, and also because it would require that we call `operator &` on the particle handler before the triangulation. The point I'm just trying to make is that, perhaps, the *exact details* of what helper classes we want to use for serialization is not something the user should need to know. Perhaps that is a point orthogonal to the first two bullet points, though.)

**Comment by blaisb** [comment]
I would like to propose something else. Maybe we could design a new tutorial from scratch that would allow us to have serialization in a "full context". Something like a transient Navier-Stokes where particles are injected as tracer particles. We could do for example flow around a 2D cylinder and use the particles to visualize the vortex shedding. This way we could have one tutorial that checkpoints solution (which are actually used because we need the previous time-steps for the flow solution) and also checkpoints particles. The tutorial would use the parallel::distributed triangulation.

**Comment by bangerth** [comment]
@blaisb Sure, even better!

**Comment by kronbichler** [comment]
That sounds like a good plan to me as well. We need to make sure that the Navier-Stokes part does not make issues too complicated, so we probably want to pick the simplest possible setup for the fluid, to concentrate on the coupling and serialization. :+1:

**Comment by pcafrica** [comment]
Hi! Any updates on this topic?
I'm willing to work on the parallel checkpointing tutorial, wondering whether anybody is willing to discuss/help.

**Comment by blaisb** [comment]
I'd be willing to discuss / help :) @pcafrica

**Comment by bangerth** [comment]
Moving to the 9.9 release.

**Comment by blaisb** [comment]
Agreed.
